code/game/machinery/slotmachine.dm

```diff
@@ -91,11 +96,8 @@
 			user << "<span class='notice'>You insert a [C.cmineral] coin into [src]'s slot!</span>"
 			balance += C.value
 			qdel(C)
-
-		return
-
-	else if(!balance) //to prevent coins from magically disappearing
-		..()
```

Comment:
instead of properly coding a way to drop coin when deconstructed, some fucko decided to make the slot machine undeconstructable as long as there's coins in it... The solution was literally 4 lines... and using a proc the original coder had already made... Seriously people.